### ISSUE TITLE
DeferredList should cancel the deferred it's waiting on even if already called.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7856]
|<img alt="rthille's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| rthille reported|
|-|-|
|Trac ID|trac#7856|
|Type|defect|
|Created|2015-04-16 00:56:06Z|

If a normal Deferred is already called, but is waiting on a Deferred returned by one of its callbacks, when cancel is called, it will instead call cancel on the Deferred upon which it is waiting.

DeferredList should do the same.

<details><summary>Searchable metadata</summary>

```
trac-id__7856 7856
type__defect defect
reporter__rthille rthille
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__new new
resolution__None None
component__core core
keywords__ 
time__1429145766905091 1429145766905091
changetime__1429249961041025 1429249961041025
version__None None
owner__None None
cc__twm
```
</details>
